packages/utils/telemetry-utils/src/test/config.spec.ts

```diff
@@ -0,0 +1,260 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+    CachedConfigProvider,
+    ConfigTypes,
+    IConfigProviderBase,
+    inMemoryConfigProvider,
+} from "../config";
+
+describe("Config", () => {
+    const getMockStore = ((settings: Record<string, string>): Storage => {
+        const ops: string[] = [];
+        return {
+            getItem: (key: string): string | null => {
+                ops.push(key);
+                return settings[key];
+            },
+            getOps: (): Readonly<string[]> => ops,
+            length: Object.keys(settings).length,
+            clear: () => { },
+            key: (_index: number): string | null => null,
+            removeItem: (_key: string) => { },
+            setItem: (_key: string, _value: string) => { },
+        };
+    });
+
+    const untypedProvider = ((settings: Record<string, ConfigTypes>): IConfigProviderBase => {
+        return {
+            getRawConfig: (name: string): ConfigTypes => settings[name],
+        };
+    });
+
+    it("Typing - storage provider", () => {
+        const settings = {
+            number: "1",
+            badNumber: "{1}",
+            stringAndNumber: "1",
+            stringAndBoolean: "true",
+            string: "string",
+            boolean: "true",
+            badBoolean: "truthy",
+            numberArray: `[1, 2, 3]`,
+            badNumberArray: `["one", "two", "three"]`,
+            stringArray: `["1", "2", "3"]`,
+            badStringArray: "1",
+            booleanArray: "[true, false, true]",
+            BadBooleanArray: "[true, 1, true]",
+        };
+
+        const mockStore = getMockStore(settings);
+        const config = new CachedConfigProvider(inMemoryConfigProvider(mockStore));
+
+        assert.equal(config.getNumber("number"), 1);
+        assert.equal(config.getNumber("badNumber"), undefined);
+        assert.equal(config.getNumber("stringAndNumber"), 1);
+
+        assert.equal(config.getString("stringAndNumber"), "1");
```

Comment:
I'd drop the "stringAnd" cases from this test, and simply test `getString` on every single setting.  As-is, it's weird to have `number: "1"` above, and only be testing `getString` on those, when the arrays are also in there as strings.

Reply:
I'd leave these cases but probably a rename is in order to iron out any confusion. The problem was that the converter would have looked at the value (`"1"`), interpret it as a primitive type and return a number. This broke the scenario of trying to read a string which was actually the serialization of a number so every time you'd try to read a string, you'd get the default value because you weren't getting the expected type. When that was fixed, these test cases have been added.

Reply:
How about this, for each case:
* getRawConfig should return the string value
* getFoo should return the value parsed as Foo
* getString should return the string value

Then remove the "stringAnd" cases as redundant.